platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -1855,6 +1858,19 @@ public ConfigurationRole getRoleAtCreation() {
         return roleAtCreation;
     }
 
+    private void assertNotDetachedExtensionDoingExtending(Iterable<Configuration> extendsFrom) {
+        if (isDetachedConfiguration()) {
+            String summarizedExtensionTargets = StreamSupport.stream(extendsFrom.spliterator(), false)
+                .map(c -> "'" + c.getName() + "'")
+                .collect(Collectors.joining(", "));
+            DeprecationLogger.deprecateAction(String.format("Calling extendsFrom on %s", this.getDisplayName()))
```

Comment:
```suggestion
            DeprecationLogger.deprecateAction("Calling extendsFrom on a detached configuration")
```

I think we should put the thing thats being deprecated here and the configuration in question in the context. We're deprecating extending any detached configuration, not specifically this one detached configuration

Reply:
Can't do this, as the `displayName` contains the `configuration:` part as a prefix.  

Detached configurations are all named `detachedConfigurationX`, and the context bit of the message says "This is a detached configuration and can not extend other configurations" so it should be clear.